lib/services/workflow-api-service.js

```diff
@@ -155,7 +237,32 @@ function workflowApiServiceFactory(
     };
 
     WorkflowApiService.prototype.findActiveGraphForTarget = function(target) {
-        return taskGraphStore.findActiveGraphForTarget(target);
+        return waterline.graphobjects.find({ node: target })
+            .then(function(workflows){
+                return _.filter(workflows, function(wf) {
+                    return wf.active();
```

Comment:
In this case though, `_status = "pending" and node = target` is the underlying criteria for an active workflow for the node, and it will only ever be zero or one objects, whereas doing a find all has no upper bound and could potentially be very slow. I get that it's easier to understand what's happening from the `return wf.active()` statement, but that was also my intent with the method name of `findActiveGraphForTarget`.

Re your second point, is there perhaps a way to cast the objects coming direct out of mongo to resemble the objects out of the waterline layer (I thought there wasn't a difference, but if there is we should probably be doing this by default)? What are the differences in the resources?